services/web-server/test/server_test.js

```diff
@@ -0,0 +1,45 @@
+const assert = require('assert');
+const helper = require('./helper');
+const request = require('superagent');
+const testing = require('taskcluster-lib-testing');
+
+suite(testing.suiteName(), () => {
+  for (let [origin, given, expected] of [
+    // a "regular" rootUrl
+    ['https://tc.example.com', 'https://tc.example.com', 'https://tc.example.com'],
+    // * is used in local development
+    ['*', 'http://localhost', 'http://localhost'],
+    [
+      // legacy rootUrl is a special case
+      'https://taskcluster.net',
+      'https://taskcluster-ui.herokuapp.com',
+      'https://taskcluster-ui.herokuapp.com',
+    ], [
+      // check that deploy previews are supported..
+      'https://tc.example.com',
+      'https://deploy-preview-897--taskcluster-web.netlify.com/',
+      'https://deploy-preview-897--taskcluster-web.netlify.com/',
+    ],
+  ]) {
+    suite(`with allowedOrigin = ${origin}, request origin = ${given}`, function() {
+      suiteSetup(async function() {
+        await helper.load('cfg');
+        helper.load.save();
+        helper.load.cfg('server.allowedOrigin', origin);
+      });
+
+      suiteTeardown(function() {
+        helper.load.restore();
+      });
+
+      helper.withServer(false, () => false);
+
+      test('request', async function() {
+        const res = await request
+          .options(`http://localhost:${helper.serverPort}/graphql`)
```

Comment:
What does `.options` do? I can't seem to find the relevant section in http://visionmedia.github.io/superagent/.

Reply:
It makes an `OPTIONS /graphql` request, similar to what a browser would do for POST (I think..).